### PR TITLE
Add `__error` to freebsd shims

### PR DESCRIPTION
**Pull request by InfRandomness**
Signed-off-by: InfRandomness <infrandomness@gmail.com>

### CODE REVIEW

File: src/shims/unix/freebsd/foreign_items.rs [hunk 1]
```diff
@@ -24,6 +24,13 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
...
                 this.write_null(dest)?;
             }
+
+            // errno
+            "__error" => {
+                let [] = this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                let errno_place = this.last_error_place()?;
+                this.write_scalar(errno_place.to_ref(this).to_scalar()?, dest)?;
+            }
             _ => return Ok(EmulateByNameResult::NotSupported),
```
Comment (RalfJung) [inline src/shims/unix/freebsd/foreign_items.rs:34]:
```suggestion

            _ => return Ok(EmulateByNameResult::NotSupported),
```
to keep it visually separated from the "errno" category

**Comment by RalfJung** [comment]
The code looks good, but there should be a test. :)
Probably one of the existing tests needs this, it just needs to be enabled for FreeBSD.

**Comment by InfRandomness** [comment]
> The code looks good, but there should be a test. :) Probably one of the existing tests needs this, it just needs to be enabled for FreeBSD.

`current_dir_with_isolation` does, but then we need a new pipeline?

**Comment by RalfJung** [comment]
Just add it to the list of tests at
https://github.com/rust-lang/miri/blob/5815d8d81cbe9b5db3d70dca1cee87d17e259afe/ci.sh#L63

**Comment by InfRandomness** [comment]
> Just add it to the list of tests at
> 
> https://github.com/rust-lang/miri/blob/5815d8d81cbe9b5db3d70dca1cee87d17e259afe/ci.sh#L63

oh, so that's what this is for !
Nice

**Comment by RalfJung** [comment]
Nice. :)
@bors r+

**Pull request merged**